x/tokenfactory/keeper/bankactions.go

```diff
@@ -2,10 +2,35 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 
 	"github.com/osmosis-labs/osmosis/v23/x/tokenfactory/types"
 )
 
+func (k Keeper) Mint(ctx sdk.Context, sender string, amount sdk.Coin, mintTo string) error {
+	// pay some extra gas cost to give a better error here.
+	_, denomExists := k.bankKeeper.GetDenomMetaData(ctx, amount.GetDenom())
+	if !denomExists {
+		return types.ErrDenomDoesNotExist.Wrapf("denom: %s", amount.GetDenom())
+	}
+
+	authorityMetadata, err := k.GetAuthorityMetadata(ctx, amount.GetDenom())
+	if err != nil {
+		return err
+	}
+
+	if sender != authorityMetadata.GetAdmin() {
+		return types.ErrUnauthorized
+	}
+
+	err = k.mintTo(ctx, amount, mintTo)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
nit: 
```suggestion
	return k.mintTo(ctx, amount, mintTo)
```

Reply:
I did this intentionally to make error handling as explicit as possible. Additionally, this ensures that we won't forget to wrap this error if we decide to do so in the future.